README.md

```diff
@@ -63,1498 +96,753 @@ Delegating a capability is like the statement `message`. Task is akin to `messag
-```
+1. Commits the Signature to the cryptographic algorithms used
+2. Describes how the payload was serialized before signing
 
-# 3 Task
+## Invocation Payload
 
-A Task is the smallest unit of work that can be requested from a UCAN. It describes one `(resource, ability, input)` triple. The `input` field is free form, and depend on the specific resource and ability being interacted with, and is not described in this specification.
-
-Using the JavaScript analogy from the introduction, a Task is similar to wrapping a call in an anonymous function:
+The Invocation Payload attaches sender, receiver, and provenance to the [Task].
+ 
+| Field       | Type    | Required           | Description |                                                                    |
+|-------------|---------|--------------------|-------------|--------------------------------------------------------------------|
+| [Issuer]    | `iss`   | `DID`              | Yes         | The DID of the [Invoker]                                           |
+| [Subject]   | `sub`   | `DID`              | Yes         | The [Subject] being invoked                                        |
+| [Audience]  | `aud`   | `DID`              | No          | The DID of the intended [Executor] if different from the [Subject] |
+| [Command]   | `do`    | `String`           | Yes         | The [Command]                                                      |
+| [Arguments] | `args`  | `{String : Any}`   | Yes         | The [Command]'s [Arguments]                                        |
+| [Proofs]    | `prf`   | `[&Delegation]`    | Yes         | [Delegation]s that prove the chain of authority                    |
+| [Metadata]  | `meta`  | `{String : Any}`   | No          | Arbitrary [Metadata]                                               |
+| [Nonce]     | `nonce` | `Bytes`            | No          | A unique, random nonce                                             |
+| [Expiry]    | `exp`   | `Integer`[^js-num] | No          | The timestamp at which the Invocation becomes invalid              |
+| [Issued At] | `iat`   | `Integer`[^js-num] | No          | The timestamp at which the Invocation was created                  |
+| [Cause]     | `cause` | `&Receipt`         | No          | An OPTIONAL CID of the [Receipt] that enqueued the [Task]          |
+
+The shape of the `args` MUST be defined by the `do` field type. This is similar to how a method or message contain certain data shapes in object oriented or actor model languages respectively. Using the JavaScript analogy from the introduction, an Action is similar to wrapping a call in a closure:
 
-```json
+```js
+// Command
 {
-  "on": "mailto:alice@example.com",
-  "call": "msg/send",
-  "input": {
-    "to": [
-      "bob@example.com",
-      "carol@example.com"
-    ],
+  "do": "msg/send",
+  "args": {
+    "from": "mailto:alice@example.com",
+    "to": [ "bob@example.com", "carol@example.com" ],
     "subject": "hello",
```

Comment:
Discussed on today's call-- someone CTRL-Fing for `subject` could get tripped up, so a quick rename to `emailSubject` or `messageTitle` or same might be a quick fix.